components/index.js

```diff
@@ -1,82 +1,49 @@
-var components = require('../components.js');
-var peerDependentsMap = null;
-
-function getPeerDependentsMap() {
-	var peerDependentsMap = {};
-	Object.keys(components.languages).forEach(function (language) {
-		if (language === 'meta') {
-			return false;
-		}
-		if (components.languages[language].peerDependencies) {
-			var peerDependencies = components.languages[language].peerDependencies;
-			if (!Array.isArray(peerDependencies)) {
-				peerDependencies = [peerDependencies];
-			}
-			peerDependencies.forEach(function (peerDependency) {
-				if (!peerDependentsMap[peerDependency]) {
-					peerDependentsMap[peerDependency] = [];
-				}
-				peerDependentsMap[peerDependency].push(language);
-			});
-		}
-	});
-	return peerDependentsMap;
-}
-
-function getPeerDependents(mainLanguage) {
-	if (!peerDependentsMap) {
-		peerDependentsMap = getPeerDependentsMap();
+const components = require('../components.js');
+const getLoad = require('../dependencies');
+
+
+/**
+ * The set of all languages which have been loaded using the below function.
+ *
+ * @type {Set<string>}
+ */
+const loadedLanguages = new Set();
+
+/**
+ * Loads the given languages and adds them to the current Prism instance.
+ *
+ * If no languages are provided, __all__ Prism languages will be loaded.
+ *
+ * @param {string|string[]} [languages]
+ * @returns {void}
+ */
+function loadLanguages(languages) {
+	if (languages === undefined) {
+		languages = Object.keys(components.languages).filter(l => l != 'meta');
+	} else if (!Array.isArray(languages)) {
+		languages = [languages];
 	}
-	return peerDependentsMap[mainLanguage] || [];
-}
 
-function loadLanguages(arr, withoutDependencies) {
-	// If no argument is passed, load all components
-	if (!arr) {
-		arr = Object.keys(components.languages).filter(function (language) {
-			return language !== 'meta';
-		});
-	}
-	if (arr && !arr.length) {
-		return;
-	}
-
-	if (!Array.isArray(arr)) {
-		arr = [arr];
-	}
+	// the user might have loaded languages via some other way or used `prism.js` which already includes some
+	// we don't need to validate the ids because `getLoad` will ignore invalid ones
+	const loaded = [...loadedLanguages, ...Object.keys(Prism.languages)];
```

Comment:
If we can get the current list of loaded languages from `Prism.langauges`, do we need to track it internally?

Reply:
Yes, we do because of extensions like CSS extras.